src/manager/tilingManager.js

```diff
@@ -44,6 +44,18 @@ var TilingManager = class TilingManager extends MsManager {
             this.ratio = schema.get_double('ratio-value');
             this.tileWindows();
         });
+        // update the layout of all workspaces if default-layout changed
+        this.observe(this.layoutsSettings, 'changed::defaultlayout', (schema) => {
+            GLib.idle_add(GLib.PRIORITY_DEFAULT, () => {
+                Me.msWorkspaceManager.msWorkspaceList.forEach(
+                    (msWorkspace) => {
+                        msWorkspace.setTilingLayout(schema.get_string('defaultlayout'));
+                    }
+                );
+                return GLib.SOURCE_REMOVE;
+            });
+        }
+        );
```

Comment:
The change of the default layout should impact only the new workspace created by assigning the default layout instead of the maximized one. (And maybe the last empty workspace?)

Reply:
done , changes will only affect the workspaces that will be created after selecting the default layout